### PR TITLE
fix(Button): loading icon appearing when rendering

### LINKED ISSUE #44379: Unwanted loading icon appearing in <Button />
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-fxkmn3)

### Steps to reproduce

1. Have a button inside `<Freeze></Freeze >` ([react-freeze](https://github.com/software-mansion/react-freeze))
2. Freeze and unfreeze
3. for few moment you'll have a loading icon in the button

### What is expected?

the button should not have the loading icon, unless specified.

### What is actually happening?

For few moment the button has a loading icon.

| Environment | Info |
| --- | --- |
| antd | 5.8.4 |
| React | 18.x |
| System | windows 10 |
| Browser | latest chrome |

### PR DESCRIPTION
**Pull request by lzl0304**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #44379

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix the loading icon appearing when rendering |
| 🇨🇳 Chinese | 修复渲染时按钮出现加载图标 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 0a3db19</samp>

Fix loading icon bug in `button` component. Only render the icon if `innerLoading` prop is true in `components/button/button.tsx`.

### <samp>🤖 Generated by Copilot at 0a3db19</samp>

* Fix loading icon bug in button component ([link](https://github.com/ant-design/ant-design/pull/44476/files?diff=unified&w=0#diff-463cd38ab465e1f3b90463a403d8e8399f926361436eaa9adb8a76a8bf9d4140L253-R255))

### CODE REVIEW

File: components/button/button.tsx [hunk 1]
```diff
@@ -250,7 +250,9 @@ const InternalButton: React.ForwardRefRenderFunction<
         {icon}
       </IconWrapper>
     ) : (
-      <LoadingIcon existIcon={!!icon} prefixCls={prefixCls} loading={!!innerLoading} />
+      innerLoading && (
+        <LoadingIcon existIcon={!!icon} prefixCls={prefixCls} loading={!!innerLoading} />
```
[the remark this answers is not in the document]
Comment (MadCcc) [reply components/button/button.tsx:254]: +1，这里直接做逻辑判断会导致动画丢失

[the remark this answers is not in the document]
Comment (lzl0304) [reply components/button/button.tsx:254]: 感谢大神指点

**Comment by MadCcc** [comment]
来个测试用例？

**Comment by vparaskevas** [comment]
Could you please apply the same fix for AntD v4?

**Pull request closed without merging**